### PR TITLE
Returning disconnection reason improved

### PR DESCRIPTION
**Pull request by philips77**
This PR improves reporting the disconnection reason. Instead of a separate disconnection callback, the `BleManagerCallbacks#onDeviceDisconnected(...)` is used. A new method with `status` has been added, and the old one deprecated. Both will be called until the deprecated one is removed. The same situation with `onLinkLossOccurred(...)`.

### CODE REVIEW

File: ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java [hunk 1]
```diff
@@ -74,8 +74,38 @@
...
+	default void onDeviceDisconnected(@NonNull final BluetoothDevice device) {
+		// do nothing
+	}
+
+	/**
+	 * Called when the device has disconnected (when the callback returned
+	 * {@link BluetoothGattCallback#onConnectionStateChange(BluetoothGatt, int status, int)} with
+	 * newstate DISCONNECTED), but ONLY if the {@link ConnectRequest#shouldAutoConnect()} method
+	 * returned false for this device when it was connecting.
+	 * Otherwise the {@link #onLinkLossOccurred(BluetoothDevice)} method will be called instead.
```
Comment (moritz-eder) [inline ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java]: I think here you want to reference the onLinkLossOccurred method with int parameter

File: ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java [hunk 2]
```diff
@@ -74,8 +74,38 @@
...
 	 */
-	void onDeviceDisconnected(@NonNull final BluetoothDevice device);
+	@Deprecated
+	default void onDeviceDisconnected(@NonNull final BluetoothDevice device) {
+		// do nothing
+	}
+
+	/**
+	 * Called when the device has disconnected (when the callback returned
+	 * {@link BluetoothGattCallback#onConnectionStateChange(BluetoothGatt, int status, int)} with
```
Comment (moritz-eder) [inline ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java]: Change(BluetoothGatt, int status, int)
should be (I think)
Change(BluetoothGatt, int, int)

Or did you add it for clarity?

Comment (philips77) [reply ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java]: I did add it for clarity, but I'm closing the PR. There will be another one with more drastically changes.

File: ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java (hunk 1 above)
Comment (philips77) [reply ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java]: True, but...

File: ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java (hunk 2 above)
Comment (philips77) [reply ble/src/main/java/no/nordicsemi/android/ble/BleManagerCallbacks.java]: See https://github.com/NordicSemiconductor/Android-BLE-Library/pull/149#issuecomment-606750235.

**Comment by philips77** [comment]
This PR has been replaced with #174.

**Pull request closed without merging**